### DISCUSSION TITLE
Could anyone help find the right add to basket container here?

### DISCUSSION THREAD

**Discussion by lozbinks** [jef/streetmerchant#2023] [Help]
Been pulling my hair out for hours over this one and I'm sure it's something completely obvious that I've missed, hopefully someone here can help me out.

So I'm monitoring the ps5 and xbox consoles on the site very.co.uk, however its never detecting the stock.

I think this is because the in stock labels ['available', 'low stock', 'pre-order'] that SM looks for are only displayed with these consoles once a certain option (i.e extra controller) is selected. (As per the images below)

As the page loads without any option selected no labels will be visible to search:
![very series s in stock](https://user-images.githubusercontent.com/78455265/108996787-be036e80-7696-11eb-932c-3fea59c86d93.JPG)

Only once a option is selected the in stock label will be shown:
![very series s option selected](https://user-images.githubusercontent.com/78455265/108998281-97463780-7698-11eb-8bc6-b45d0ca50046.JPG)

Unless there is anyway to allow streetmerchant to check for the hidden in stock labels ['available', 'low stock', 'pre-order now'] without an option being selected I assume the best thing would be to just check for the add to basket button.

However I have tried loads of what appear to be the addToBasketButton selectors in my store ts file to no avail.

Here is my current store TS file for very.co.uk:

```typescript
import {Link, Store} from './store';
import {logger} from '../../logger';
import {parseCard} from './helpers/card';

export const Very: Store = {
	currency: '£',
	labels: {
		inStock: {
			container: '.stockMessaging .indicator',
			text: ['available', 'low stock', 'Pre-order now']
		},
		maxPrice: {
			container: '.priceNow',
			euroFormat: false // Note: Very uses non-euroFromat as price seperator
		},
		outOfStock: {
			container: '.stockMessaging .indicator',
			text: ['Unavailable']
		}
	},
	links: [
		{
			brand: 'test:brand',
			model: 'test:model',
			series: 'test:series',
			url:
				'https://www.very.co.uk/playstation-5-playstation-5-dualsensenbspwireless-controller/1600485234.prd'
		},
		{
			brand: 'sony',
			model: 'ps5 console',
			series: 'sonyps5c',
			url:
				'https://www.very.co.uk/playstation-5-disc-console-with-optional-extras/1600568976.prd'
		},
				{
			brand: 'sony',
			model: 'ps5 digital',
			series: 'sonyps5de',
			url: 'https://www.very.co.uk/playstation-5-digital-console-with-optional-extras/1600549412.prd'
		},
		{
			brand: 'microsoft',
			model: 'xbox series x',
			series: 'xboxsx',
			url:
				'https://www.very.co.uk/xbox-series-x-xbox-series-x-withnbspoptional-extras/1600503631.prd'
		},
		{
			brand: 'microsoft',
			labels: {
				inStock: {
					container: '#addToBasketButton',
					text: ['add to basket']
				}
			},model: 'xbox series s',
			series: 'xboxss',
			url:
				'https://www.very.co.uk/xbox-series-s-xbox-series-s-with-optional-extras/1600503630.prd'

			}
	],
	name: 'very'
};
```

The product URL for the example in this article is:
https://www.very.co.uk/xbox-series-s-xbox-series-s-with-optional-extras/1600503630.prd

Again any help would be greatly appreciated, especially @jef if you have a spare minute.

[Update] The product mentioned above is now OOS and the retailer removes the product pages when OOS however this currently in stock product below has the same selectable options required as above to make the in stock labels appear visible for anyone who may be able to help out.

https://www.very.co.uk/xbox-series-x-wireless-headset/1600573663.prd

**Comment by jef** [comment]
> Unless there is anyway to allow streetmerchant to check for the hidden in stock labels ['available', 'low stock', 'pre-order now'] without an option being selected I assume the best thing would be to just check for the add to basket button.

Hey @lozbinks, great question.

If I understand correctly, this basically what you're asking, right?

The current containers in the `very.ts` work correctly with very.co.uk/xbox-series-x-wireless-headset/1600573663.prd.

As for selecting an option so we can see if it's available or not, that isn't possible right now, but could be implemented to run a script beforehand. This is interesting as it would expand other options in the future. I'll keep you posted.

That being said, I think checking the `addToBasketButton` appears when something is in or out of stock, so unfortunately that label won't work.

**Reply by lozbinks** [reply]
Hey @jef!

Thanks for the reply, hope your doing well.


I did actually finally get it all sorted and working in the end, just forgot to update the post.

What I can confirm worked for me in the end and caught the most recent Xbox series X drop around a week ago was the following:

`labels: {
		inStock: {
			container: '#productCallToAction',
			text: ['add to basket']
		},
		maxPrice: {
			container: '.priceNow',
			euroFormat: false // Note: Very uses non-euroFromat as price seperator
		},
		outOfStock: {
			container: '.stockMessaging .indicator',
			text: ['Unavailable']
		}`
		
However I must add Very.co.uk do love to give out temporary IP bans as I've found out, so for anyone else who see's this and is planning on tracking items on very.co.uk make sure to set a longer wait time for your checks.

Once again thank you Jef for creating such a awesome tool, your work and dedication is greatly appreciated 👍

**Reply by jef** [reply]
Glad you scored yourself one! Thanks for the kind words ☺️
